pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -319,6 +320,23 @@ func (s *Service) getOrCreateAPILoadBalancer(openStackCluster *infrav1.OpenStack
 				record.Eventf(openStackCluster, "OctaviaProviderNotFound", "Provider %s specified for Octavia not found, using the default provider.", openStackCluster.Spec.APIServerLoadBalancer.Provider)
 			}
 		}
+		if openStackCluster.Spec.APIServerLoadBalancer.Flavor != nil {
+			flavors, err := s.loadbalancerClient.ListLoadBalancerFlavors()
```

Comment:
I was going to ask: why don't we filter based on name here rather than fetching all flavors and filtering locally.

The answer is: although [Octavia supports this](https://docs.openstack.org/api-ref/load-balancer/v2/#filtering), it seems that [Gophercloud does not](https://github.com/gophercloud/gophercloud/blob/59af46043901fed92a8d3de89d5d2c745477c2a9/openstack/loadbalancer/v2/flavors/requests.go#L17-L20) (yet). `Fields` in `ListOpts` only controls which fields are returned. We can't specify any filtering.

Could you add a comment here explaining that?

If that comment linked to a Feature Request in gophercloud, that would be even better. This is likely not hard to implement.

For now, though, I suspect (perhaps naively) that nobody has a deployment with so many load balancer flavors that this would become a problem.

Reply:
I added a comment with a link to feature request in Gophercloud repository:
https://github.com/gophercloud/gophercloud/issues/3049